ansible_events_ui/api/__init__.py

```diff
@@ -333,6 +226,226 @@ async def deactivate(activation_instance_id: int):
+            f"/job_instance/{job_instance_id}",
+            json.dumps(["Stdout", {"stdout": event_data.get("stdout")}]),
+        )
+
+    query = insert(models.job_instance_events).values(
+        job_uuid=event_data.get("job_id"),
+        counter=event_data.get("counter"),
+        stdout=event_data.get("stdout"),
+    )
+    await db.execute(query)
+    await db.commit()
+
+
+async def handle_jobs(data: dict, db: AsyncSession):
+    query = insert(models.job_instances).values(uuid=data.get("job_id"))
+    result = await db.execute(query)
+    (job_instance_id,) = result.inserted_primary_key
+
+    activation_instance_id = int(data.get("ansible_events_id"))
+    query = insert(models.activation_instance_job_instances).values(
+        job_instance_id=job_instance_id,
+        activation_instance_id=activation_instance_id,
+    )
+    await db.execute(query)
+    await db.commit()
+    await updatemanager.broadcast(
+        f"/activation_instance/{activation_instance_id}",
+        json.dumps(["Job", {"job_instance_id": job_instance_id}]),
+    )
+    await updatemanager.broadcast(
+        "/jobs",
+        json.dumps(["Job", {"id": job_instance_id}]),
+    )
+
+
+async def handle_actions(data: dict, db: AsyncSession):
+    logger.info(f"Start to handle actions: {data}")
+    activation_id = int(data.get("activation_id"))
+
+    if activation_id:
+        action_name = data.get("action")
+        playbook_name = data.get("playbook_name")
+        job_id = data.get("job_id")
+        fired_date = data.get("run_at")
+        status = data.get("status")
+
+        if job_id:
+            sel = (
+                select(
+                    models.activation_instances.c.id.label(
+                        "activation_instance_id"
+                    ),
+                    models.rulesets.c.id.label("ruleset_id"),
+                    models.rules.c.id.label("rule_id"),
+                    models.rules.c.name,
+                    models.rules.c.action.label("definition"),
+                    models.job_instances.c.id.label("job_instance_id"),
+                    cast(fired_date, postgresql.VARCHAR).label("fired_date"),
+                    cast(status, postgresql.VARCHAR).label("status"),
+                )
+                .join(
+                    models.rulesets,
+                    models.rulesets.c.rulebook_id
+                    == models.activation_instances.c.rulebook_id,
+                )
+                .join(
+                    models.rules,
+                    models.rules.c.ruleset_id == models.rulesets.c.id,
+                )
+                .join(
+                    models.job_instances,
+                    models.job_instances.c.uuid
+                    == cast(job_id, postgresql.UUID),
+                )
+                .where(
+                    and_(
+                        models.activation_instances.c.id == activation_id,
+                        models.rules.c.action[action_name] is not None,
```

Comment:
Have you checked the actual query generated? This expression is always evaluated as `TRUE`.

In Python `is` and `is not` operators cannot be overloaded, therefore it's cannot be used for query condition generation.
You should use 

```
models.rules.c.action[action_name].is_not(None)
```